### PR TITLE
Change the fee store v2

### PR DESCRIPTION
**Pull request by hskim881028**
Change the fee store v2

- This is a PR to change the fee storage v2 in preparation for the reboot arena
- Buy11 -> Buy12
- CombinationEquipment12
- ItemEnhancement11

change the fee storage v1 : https://github.com/planetarium/lib9c/pull/1069

### CODE REVIEW

**Review by Unengine — APPROVED** [review]

File: Lib9c/Action/CombinationEquipment.cs [hunk 1]
```diff
@@ -340,9 +333,18 @@ public override IAccountStateDelta Execute(IActionContext context)
             // Transfer Required NCG
             if (costNCG > 0L)
             {
+                var arenaSheet = states.GetSheet<ArenaSheet>();
+                if (arenaSheet.GetRowByBlockIndex(context.BlockIndex) == null)
```
Comment (ipdae) [inline Lib9c/Action/CombinationEquipment.cs:337]: is this necessary? i can't find any using row.

File: Lib9c/Action/ItemEnhancement.cs [hunk 2]
```diff
@@ -282,7 +279,15 @@ public override IAccountStateDelta Execute(IActionContext context)
             var requiredNcg = row.Cost;
             if (requiredNcg > 0)
             {
-                states = states.TransferAsset(ctx.Signer, BlacksmithAddress, states.GetGoldCurrency() * requiredNcg);
+                var arenaSheet = states.GetSheet<ArenaSheet>();
+                if (arenaSheet.GetRowByBlockIndex(context.BlockIndex) == null)
```
Comment (ipdae) [inline Lib9c/Action/ItemEnhancement.cs:283]: is this necessary? i can't find any using row.

File: Lib9c/Action/ItemEnhancement.cs [hunk 3]
```diff
@@ -23,14 +25,14 @@
...
-    /// Updated at https://github.com/planetarium/lib9c/pull/957
+    /// Updated at https://github.com/planetarium/lib9c/pull/1164
     /// </summary>
     [Serializable]
     [ActionType("item_enhancement11")]
     public class ItemEnhancement : GameAction
     {
-        public static readonly Address BlacksmithAddress = Addresses.Blacksmith;
-
+        public static Address GetFeeStoreAddress(int championshipId, int round) =>
```
[the remark this answers is not in the document]
Comment (hskim881028) [reply Lib9c/Action/ItemEnhancement.cs]: 👍🏻. fixed it. please check again 🏹

File: Lib9c/Action/CombinationEquipment.cs (hunk 1 above)
Comment (hskim881028) [reply Lib9c/Action/CombinationEquipment.cs]: 👍🏻. fixed it. please check again 🏹

File: Lib9c/TableData/ArenaSheet.cs [hunk 4]
```diff
@@ -112,8 +112,22 @@ protected override void AddRow(int key, Row value)
...
+        {
+            var roundList = OrderedList.SelectMany(row => row.Round);
+
+            if (roundList == null || !roundList.Any())
+            {
+                throw new SheetRowNotFoundException(nameof(ArenaSheet), $"BlockIndex : {blockIndex}");
+            }
+
+            var round = roundList.First(e => e.IsTheRoundOpened(blockIndex));
+            if (round == null)
```
Comment (ipdae) [inline Lib9c/TableData/ArenaSheet.cs:125]: i think we can 't handle this because `.First` will throw `InvalidOperationException`.

**Review by ipdae — APPROVED** [review]

**Pull request merged**